### PR TITLE
Allow users to request database certificates in Machine ID

### PR DESCRIPTION
**Pull request by timothyb89**
This adds a new destination-level config option to `tbot.yaml` that allows users to request database access for a particular destination certificate. Behind the scenes, this triggers generates two impersonated certs: the first identity is generated to request the specified roles and resolve the database config, then is replaced with a new identity using those roles plus a fully-formed `RouteToDatabase` request. We don't match `tsh`'s behavior here as the additional certs generated conceptually similar enough to Machine ID's multiple destinations - if users don't want a cert with `RouteToDatabase` they can just add another destination with the same roles.

Database requests are made by adding a destination entry like the following to `tbot.yaml`:
```yaml
destinations:
  - directory: ./tbot-user

    database:
      service: <teleport database name>
      username: <database username>
      database: <database name>

    # The certs won't be very useful without TLS
    kinds: [tls]
```

We currently don't support proxying or generating db-specific config files. See #11596 for an `identityfile` implementation that can help. Additionally, we don't currently have a `tsh proxy` equivalent so the legacy MySQL handlers need to be enabled, which is not the case by default in Teleport 9. However, that same PR can export a tsh-compatible identity file for the datbase identity which can be passed along to `tsh -i ...` and presumably used with its proxy commands. We plan to follow-up with another PR to improve this UX.

If using MySQL/MariaDB with the legacy proxy handler exposed, this command should connect to the database:

```bash
mariadb -vvvv --ssl-key ./tbot-user/key --ssl-ca tbot-user/tlscacerts-test --ssl-cert ./tbot-user/tlscert  -h <teleport proxy host> --port=3036 --protocol TCP -u <database username> --ssl-verify-server-cert
```

---

Somewhat unrelatedly, this PR also moves most cert renewal-related code into a new file. It makes for an unpleasant diff, but it needed to happen eventually.

### CODE REVIEW

**Comment by timothyb89** [comment]
While writing the PR description I realized I did still have a manual hack in place for the TLS CA certs. We still use `$` as a separator character since we internally need both Teleport's user and host certs, but obviously literally nothing supports this "format". I'm gonna think on the best way around this - we might just not want to write these weird cert types to destination dirs since the bot doesn't even use them, and separately write sensible CA certs.

File: tool/tbot/config/config_destination.go [hunk 1]
```diff
@@ -21,13 +21,35 @@ import (
 	"github.com/gravitational/trace"
 )
 
+// DatabaseConfig is the config for a database access request.
+type DatabaseConfig struct {
+	// Service is the service name of the
+	Service string `yaml:"service,omitempty"`
```
Comment (zmb3) [inline tool/tbot/config/config_destination.go:27]: The godoc is incomplete here. What is service supposed to be?

File: tool/tbot/renew.go [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+		AuthServers: []utils.NetAddr{*authAddr},
+		Log:         log,
+	}
+
+	c, err := authclient.Connect(ctx, authClientConfig)
+	return c, trace.Wrap(err)
+}
+
+// describeTLSIdentity writes an informational message about the given identity to
+// the log.
```
Comment (zmb3) [inline tool/tbot/renew.go:102]: This godoc looks incorrect. I don't see anything being written to any log here.

(same goes for ssh identity below)

File: tool/tbot/testhelpers/srv.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	"github.com/gravitational/teleport/lib/config"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/utils"
+	botconfig "github.com/gravitational/teleport/tool/tbot/config"
+	"github.com/gravitational/teleport/tool/tbot/identity"
+	"github.com/stretchr/testify/require"
+)
+
+func DefaultConfig(t *testing.T) *config.FileConfig {
+	t.Helper()
```
Comment (zmb3) [inline tool/tbot/testhelpers/srv.go:40]: Not really necessary since this helper cannot fail.

File: tool/tbot/renew.go (hunk 2 above)
Comment (timothyb89) [reply tool/tbot/renew.go:102]: Ah, yep, that never got updated. Will fix!

**Comment by timothyb89** [comment]
After thinking on it I think I'd like to resolve the missing TLS CAs as part of #11596 where we'll be making significant changes to artifact saving/loading anyway. Any changes here are just going to create conflicts and I don't think it's too awful to borrow the cluster's CA cert from tsh or cut and paste it from our `tlscacerts` for testing purposes.

**Review by r0mant — APPROVED** [review]

File: tool/tbot/renew.go [hunk 4]
```diff
@@ -0,0 +1,631 @@
...
+	), nil
+}
+
+// identityConfigurator is a function that alters a cert request
+type identityConfigurator = func(req *proto.UserCertsRequest)
+
+// generateIdentity uses an identity to retrieve another possibly impersonated
+// identity. The `configurator` function, if not nil, can be used to add
+// additional requests to the certificate request, for example to add
+// `RouteToDatabase`` and similar fields, however in that case it must be
```
[the remark this answers is not in the document]
Comment (timothyb89) [reply tool/tbot/renew.go:186]: whoops, thanks!

**Review by zmb3 — APPROVED** [review]

**Pull request merged**